supfile.go

```diff
@@ -13,21 +13,66 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+// EnvVar represents an environment variable
+type EnvVar struct {
+	Key   string
+	Value string
+}
```

Comment:
I think you can mostly achieve the same thing with composing `command`s in `target`s, right? I can definitely see the use for several `run`s, but don't see the need for several `upload`s.

`run` could be done like this and you could have several shell commands in a row: https://gist.github.com/stengaard/4cd2a88693df2a3b54d3